Eatery/Controllers/BRB/BRBAccountManager.swift

```diff
@@ -0,0 +1,277 @@
+                print(error)
+                self.delegate?.brbConnectionHandlerDelegateDidFailLogin(with: error.localizedDescription)
+            } else {
+                if self.failedToLogin() {
+                    self.delegate?.brbConnectionHandlerDelegateDidFailLogin(with: "Incorrect netid and/or password")
+                }
+            }
+            self.loginCount += 1
+        }
+    }
+    
+    func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
+        self.getStageAndRunBlock {
+            switch self.stage {
+            case .loginFailed:
+                self.delegate?.brbConnectionHandlerDelegateDidFailLogin(with: "Incorrect netid and/or password")
+            case .loginScreen:
+                if self.loginCount < 1 {
+                    self.login()
+                }
+            case .finished(let sessionId):
+                self.delegate?.brbConnectionHandlerDelegateDidRetrieve(sessionID: sessionId)
+            default: break
+            }
+        }
+    }
+    
+    func webView(_ webView: WKWebView, didFailProvisionalNavigation navigation: WKNavigation!, withError error: Error) {
+        self.delegate?.brbConnectionHandlerDelegateDidFailLogin(with: error.localizedDescription)
+    }
+
+    /// Gets the stage enum for the currently displayed web page and runs a
+    /// block after fetching the HTML for the page. Does not guarantee
+    /// Javascript will finish running before the block is executed.
+    func getStageAndRunBlock(block: @escaping () -> ()) {
+        getHTML(block: { (html: String) -> () in
+            if self.failedToLogin() {
+                self.stage = .loginFailed
+            } else if html.contains("<h2>Login OK</h2>") {
+                self.stage = .transition
+            } else if html.contains("<h1>CUWebLogin</h1>") {
+                if self.loginCount < 1 {
+                    self.stage = .loginScreen
+                } else {
+                    self.stage = .loginFailed
+                }
+            } else if self.url?.absoluteString.contains("sessionId") ?? false {
+                guard let url = self.url, let urlComponents = URLComponents(url: url, resolvingAgainstBaseURL: true) else {
+                    self.stage = .loginFailed
+                    block()
+                    return
+                }
+
+                if let sessionId = urlComponents.queryItems?.first(where: { $0.name == "sessionId" })?.value {
+                    self.stage = .finished(sessionId: sessionId)
+                } else {
+                    self.stage = .loginFailed
+                }
+            } else {
+                self.stage = .transition
+            }
+            
+            // Run block for stage
+            block()
+        })
+    }
+}
+
+// MARK: - Account Manager
+
+protocol BRBAccountManagerDelegate {
+    func brbAccountManagerDidFailToQueryAccount(with error: String)
+    func brbAccountManagerDidQuery(account: BRBAccount)
```

Comment:
Similar comment as @danielVebman these names can probably be shorter and more swifty

Reply:
Maybe:

    brbAccountManager(didQuery account: BRBAccount)